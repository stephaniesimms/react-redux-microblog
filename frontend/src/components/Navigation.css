.Jumbotron {
  background: #EB5757;  /* fallback for old browsers */
  background: -webkit-linear-gradient(to right, #EB5757, #000000);  /* Chrome 10-25, Safari 5.1-6 */
  background: linear-gradient(to right,  #EB5757, #000000); /* W3C, IE 10+/ Edge, Firefox 16+, Chrome 26+, Opera 12+, Safari 7+ */
}

Nav a { 
  margin-right: 2em;
  color: #e7e8e1; 
}

Nav a.active { 
  font-weight: bold; 
}