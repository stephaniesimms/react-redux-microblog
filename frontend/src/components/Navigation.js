import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import { Navbar, Nav, Jumbotron } from 'react-bootstrap';
import './Navigation.css';

/** Displays main nav and header  
 * 
 * - nav links contain routes to:
 *   - homepage
 *   - new form
 *   - individual posts
*/
class Navigation extends Component {
  render() {
    return (
      <Jumbotron className='Jumbotron'>
        <div className='ml-3'>
          <h1>Microblog</h1>
          <p className='lead'>Cooking with hellfire</p>
        </div>
        <Navbar>
          <Nav>
            <NavLink exact to='/'>Blog</NavLink>
            <NavLink exact to='/new'>Add a new post</NavLink>
          </Nav>
        </Navbar>
      </Jumbotron>
    );
  }
}

export default Navigation;