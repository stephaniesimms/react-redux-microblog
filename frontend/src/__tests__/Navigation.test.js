import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Navigation from '../components/Navigation';

it('renders without crashing', function () {
  shallow(<Navigation />);
});

it('matches snapshot', function () {
  let wrapper = shallow(<Navigation />);
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});