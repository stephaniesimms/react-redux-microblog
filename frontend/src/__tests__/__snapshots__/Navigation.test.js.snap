// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<Bootstrap(Jumbotron)
  className="Jumbotron"
>
  <div
    className="ml-3"
  >
    <h1>
      Microblog
    </h1>
    <p
      className="lead"
    >
      Cooking with hellfire
    </p>
  </div>
  <Bootstrap(undefined)>
    <Nav
      fill={false}
      justify={false}
    >
      <NavLink
        exact={true}
        to="/"
      >
        Blog
      </NavLink>
      <NavLink
        exact={true}
        to="/new"
      >
        Add a new post
      </NavLink>
    </Nav>
  </Bootstrap(undefined)>
</Bootstrap(Jumbotron)>
`;
